next.clayui.com/content/docs/components/modal.mdx

```diff
@@ -29,6 +32,147 @@ You may want to compose your content and customize with your use cases or add yo
 
 <Modal />
 
+### useModal hook
+
+The `useModal` [hook](https://reactjs.org/docs/hooks-intro.html) is required to create a Modal component in Clay, it is responsible for communicating with ClayModal and controlling states to make animations, this is necessary because Modal needs animation when opening and closing so we need to delay calls of the `onClose` so you can safely control Modal's visibility or do other things.
+
+`useModal` returns a signature with `observer` and `onClose`:
+
+- **observer** - Observer is the communication channel that connects ClayModal with `useModal`, you can call it a tunnel.
+- **onClose** - Callback is called after Modal's close animation.
+
+```jsx
+const { observer, onClose } = useModal({
+    onClose: () => setVisible(false)
+});
+```
+
+<div class="clay-site-alert alert alert-warning">
+    To avoid future problems, do not use the information or change the values ​​of `observer`, it is considered a bad practice.
```

Comment:
> "do not use the information"

Is this accurate? Makes sense to discourage the changing of the values, but reading the values doesn't seem like a bad pattern.

Reply:
I didn't want to encourage reading these values, because we can change the information contained in `observer`, so I've been thinking of providing an API to` useModal` to address the need to know when Modal is really visible after the animation is done.

Reply:
yeah thats a good idea to add an API, I don't think we should dissuade people from reading the values unless we have another means of reading that information

Reply:
Yeah, we can add this and wait for the release of Clay 3.0 officially, I'll send a PR later adding this API. But still I would like to keep this information.

Reply:
sounds good. feel free to merge